site/app/controllers/HomePageController.php

```diff
@@ -85,6 +84,7 @@ public function changeUserName(){
      * Display the HomePageView to the student.
      *
      * @Route("/home")
+     * @Route("/api/courses")
      */
     public function showHomepage() {
```

Comment:
It seems odd to have /api/course map to a function that is called showHomepage().
It would be more appropriate to have a getCourses() function that is used by showHomepage() and getApiCourses() (name could be improved for the api function). The getApiCourses() would be where the api route is mapped.
Lets try to reuse all the logic of getting a course in getCourses() and leave the formatting showHomepage and getApiCourses.

Reply:
That makes more sense. Thank you for your suggestion. Will make the change :)